server/integration-tests/src/test/java/io/smallrye/graphql/tests/client/typesafe/subscription/TypesafeClientSubscriptionTest.java

```diff
@@ -0,0 +1,91 @@
+package io.smallrye.graphql.tests.client.typesafe.subscription;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import io.smallrye.graphql.client.typesafe.vertx.VertxTypesafeGraphQLClientBuilder;
+
+@RunWith(Arquillian.class)
+@RunAsClient
+public class TypesafeClientSubscriptionTest {
+
+    @Deployment
+    public static WebArchive deployment() {
+        return ShrinkWrap.create(WebArchive.class, "dynamic-client-subscription-test.war")
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml")
+                .addClasses(SubscriptionApi.class);
+    }
+
+    @ArquillianResource
+    URL testingURL;
+
+    private static SubscriptionClientApi client;
```

Comment:
```suggestion
    private SubscriptionClientApi client;
```

... or do you mean the `prepare`/`cleanup` methods to also be `static` and `@BeforeClass`/`@AfterClass`?